### PR TITLE
Updated modules

### PR DESCRIPTION
**Pull request by wardhaddadin1**
Updated modules with recent fixes/improvements

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,39 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
...
+
+## Added
+
+- updated to use `uv`.
+- Warning if `load_from_dict` is passed a dictionary with arbitrary keys outside of the expected specification that are ignored when loading a representation.
+- Added `map_light` features, which are a combination of Morgan, Avalon, Reduced Graph and handcrafted descriptors from `rdkit`.
+
+- Added uncertainty support (`predict_with_std`, `predict_with_prediction_interval`, and `sample`) for the `ensemble_regressor` model.
+- Added an `average_features_regressor` model that predicts based on the average of the input model features
+- Added GammaConformityScore and ResidualNormalisedScore to mapie_regressor. These should allow for more adaptive prediction intervals
```
Comment (shug3502) [inline CHANGELOG.md:34]: Nitpick: include backquotes `` for GammaConformityScore ResidualNormalisedScore and mapie_regressor?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,39 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 ## [Unreleased]
 
 ---------------------------------------------------------
+## [0.6.0] - 2024-08-23
+
+## Removed
+
+- removed support for python 3.8 and 3.9
```
Comment (shug3502) [inline CHANGELOG.md:16]: 🎉

File: init_conda_venv.sh [hunk 3]
```diff
@@ -10,7 +10,13 @@ echo "Setting up virtualenv"
 
 # Make sure to bypass a possible $PIP_REQUIRE_VIRTUALENV=true set by users
 # This is because pip is unable to detect that it is inside conda's virtual environment - and would throw an error
-env PIP_REQUIRE_VIRTUALENV=false conda env create -f environment.yaml --prefix .venv
+env PIP_REQUIRE_VIRTUALENV=false ${CONDA_OR_MAMBA} \
```
Comment (shug3502) [inline init_conda_venv.sh:13]: Question: Looks like CONDA_OR_MAMBA is not set here? Is it available indirectly from somewhere else?

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,39 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
...
+
+- updated to use `uv`.
+- Warning if `load_from_dict` is passed a dictionary with arbitrary keys outside of the expected specification that are ignored when loading a representation.
+- Added `map_light` features, which are a combination of Morgan, Avalon, Reduced Graph and handcrafted descriptors from `rdkit`.
+
+- Added uncertainty support (`predict_with_std`, `predict_with_prediction_interval`, and `sample`) for the `ensemble_regressor` model.
+- Added an `average_features_regressor` model that predicts based on the average of the input model features
+- Added GammaConformityScore and ResidualNormalisedScore to mapie_regressor. These should allow for more adaptive prediction intervals
+
+- Added `out_of_sample_r2` metric
```
Comment (shug3502) [inline CHANGELOG.md:36]: Suggestion: specify regression metric?

**Comment by shug3502** [comment]
Awesome work @wardhaddadin1 🥇 - all those type annotations updated too! My questions/comments are minor ones.

**Review by shug3502 — APPROVED** [review]

File: init_conda_venv.sh (hunk 3 above)
Comment (wardhaddadin1) [reply init_conda_venv.sh:13]: good spot! I copy and pasted only half the file

**Pull request merged**